### DISCUSSION TITLE
Question about PnP Search Box Dynamic Suggestions

### DISCUSSION THREAD

**Discussion by Kotola** [microsoft-search/pnp-modern-search#3288] [General]
Hi Community! 
After done hours of testing, googling, bingling and AI Chating (got to use ChatGPT and BCE cos my colleagues are tired of replying to my question bombing (LOL)) I came here to ask about the feature does it exist or not - are there Dynamic Suggestions in SP PnP Modern Search Search Box Web Part like there are in the original SharePoint search? Me and my customer haven't found any documentation nor facts about this. With Dynamic Suggestions I mean those suggested entries for search queries which appear  immediately when user start to type search query in SharePoint or Microsoft 365 search box in the title bar.

Thank you in advance your replies and references you are going to post here! Appreciate your work a lot!

**Comment by kasperbolarsen** [comment]
I guess you are referring to something like this?
https://microsoft-search.github.io/pnp-modern-search/usage/search-box/#query-suggestions
which can be modified to fit specific requirements using this:
 https://microsoft-search.github.io/pnp-modern-search/extensibility/custom_suggestions_provider/

**Reply by Kotola** [reply]
Yes, both of these topics are familiar to, but this is not exactly what I'm looking for. I'm interested in to know if Search Box Web Part is capable of giving suggestions based on first letters user is typing in to the Search Box like SharePoint and Microsoft 365 Start Page does in their Search Boxes as shown in this image. 
![image](https://github.com/microsoft-search/pnp-modern-search/assets/35297179/18f56674-6b6f-46ea-8e1f-549a86da13be)

I'm satisfied with simple response 'Yes' or 'No'.
